accepted/2020/form-factors.md

```diff
@@ -0,0 +1,206 @@
+
+We are intentionally not trying to address the subsets of incompatible APIs by introducing new smaller TFMs that exclude these APIs. Building systems with walls leads to ecosystem fragmentation. We have done that in the past and we were not happy with the result, like for example with .NET Framework 4.0 and the "client profile" concept. Our focus is going to be on exposing a single set of broadly available and useful APIs defined by the [`net5` TFM](https://github.com/dotnet/designs/blob/master/accepted/2020/net5/net5.md) introduced with .NET 5.0 and future versions of the platform that expand it.
+
+### IL Linking
+
+The [IL linker](https://github.com/mono/linker) can generate optimized applications that fail at runtime, due to not being able to reliably analyze various problematic code patterns (largely centered on reflection use). It is not possible or desirable to simply invest more to improve the linker, as a strategy. Instead, build-time source generators will be the recommended mitigation for arbitrary reflection use.
+
+The barrier for entry for building source generators is very high today. There is no standard solution and only a few libraries can afford to build a custom one. Razor pages, gRPC, WPF data binding, and Data-Contract serializers each have their own custom build-time source generator solutions today. First-class Roslyn support for [source generators](https://github.com/dotnet/roslyn/blob/master/docs/features/source-generators.md) will lower the barrier of entry for components that need to generate source code.
+
+The existing annotations that provide hints to the IL linker about what to keep or what is safe to link are going to stay, but their use is going to be less preferred as compared to source generators. They provide a less predictable experience that is hard for users to get right.
+
+We will introduce a linker compatibility analyzer to detect code patterns that cannot be reliably analyzed by the linker. This will still be needed for libraries that do not or cannot adopt the source generation technique.
+
+### Single File
+
+The single-file form factor bundles all code into a single binary that the code is directly executed from. Single-file apps are easier to deploy and distribute. Many users have asked for this form factor. We will focus on producing traditional immutable single-file executables, and provide guidelines for application developers to target to successfully target the single-file form factor.
+
+We will not attempt to re-create application virtualization solutions such as [BoxedApp](https://www.boxedapp.com/docs/index.html) or (`App-V`)[https://docs.microsoft.com/en-us/windows/application-management/app-v/appv-for-windows] that can make unaware applications deployable as single-file executables by virtualizing the operating system. Application virtualization solutions are a unique domain of sophisticated technology, outside the realm of .NET runtime.
```

Comment:
The App-V link currently renders like this:

(`App-V`)[https://docs.microsoft.com/en-us/windows/application-management/app-v/appv-for-windows]

The square braces and parens should be swapped:

```
[`App-V`](https://docs.microsoft.com/en-us/windows/application-management/app-v/appv-for-windows)
```

Which will render like this:

[`App-V`](https://docs.microsoft.com/en-us/windows/application-management/app-v/appv-for-windows)